src/spaces/utilities.jl

```diff
@@ -43,6 +43,31 @@ function euclidean_distance(
     sqrt(sum(min.(direct, spacesize(space) .- direct).^2))
 end
 
+function euclidean_distance(
+    p1::ValidPos,
+    p2::ValidPos,
+    space::Union{ContinuousSpace{D,P},AbstractGridSpace{D,P}}
+) where {D,P}
+    s = spacesize(space)
+    distance_squared = 0.0
```

Comment:
is this better?

```suggestion
    distance_squared = zero(eltype(p1))
```

Reply:
`zero(float(eltype(p1))` should be the right version I guess

Reply:
actually I don't think so, since if p1 is a tuple of ints then it is better to have 0 instead of 0.0